client/src/components/new-request/main-container.vue

```diff
@@ -5,49 +5,62 @@
       <v-col cols="auto" v-if="displayedComponent !== 'Success'">
         <button id="back-to-search-btn"
                 class="modal-activator pa-0 link-std-sans-ul"
-                @click="cancelAnalyzeName">
+                @click="cancelAndResetState">
           <span class="link-std-sans-ul">
             <v-icon class="ma-0 pa-0 mr-n1 mini-back-arrow">arrow_back_ios</v-icon>
             {{ editMode ? 'Return' : 'Start Search Over' }}
           </span>
         </button>
+        <countdown-timer :timerName="timerName" colorString="#1669bb" bgColorString="#efefef" />
       </v-col>
     </v-row>
     <slot name="content"/>
   </v-container>
 </template>
 
 <script lang="ts">
-import { Component, Vue } from 'vue-property-decorator'
-import newReqModule, { EXISTING_NR_TIMER_NAME } from '@/store/new-request-module'
-import timerModule from '@/modules/vx-timer'
+import { Component, Mixins, Watch } from 'vue-property-decorator'
+
+import DisplayedComponentMixin from '@/components/mixins/displayed-component-mixin'
+import SessionTimerMixin from '@/components/session-timer/session-timer-mixin'
+import CountdownTimer from '@/components/session-timer/countdown-timer.vue'
+
+import newReqModule, {
+  EXISTING_NR_TIMER_NAME, EXISTING_NR_TIMEOUT_MS,
+  NR_COMPLETION_TIMER_NAME, NR_COMPLETION_TIMEOUT_MS
+} from '@/store/new-request-module'
+import timerModule from "@/modules/vx-timer"
+
+@Component({
+  components: {
+    CountdownTimer
+  }
+})
+export default class MainContainer extends Mixins(SessionTimerMixin, DisplayedComponentMixin) {
+  componentName: string = ''
+  displayTimer: boolean = false
+  timerName: string = ''
+  countdownMins: number = 0
 
-@Component({})
-export default class MainContainer extends Vue {
   get editMode () {
     return newReqModule.editMode
   }
-  async cancelAnalyzeName () {
-    if (this.editMode) {
-      // Check in the NR to release the INPROGRESS lock on the NR
-      await newReqModule.cancelEditExistingRequest()
-      await newReqModule.checkinNameRequest()
-      timerModule.stopTimer({ id: EXISTING_NR_TIMER_NAME })
-      // Redirect to the start
-      // Catch any errors, so we don't get errors like:
-      // Avoided redundant navigation to current location: "/"
-      await this.$router.replace('/').catch(() => {})
-    } else {
-      await newReqModule.cancelAnalyzeName('Tabs')
-      // Redirect to the start
-      // Catch any errors, so we don't get errors like:
-      // Avoided redundant navigation to current location: "/"
-      await this.$router.replace('/').catch(() => {})
+
+  @Watch('componentName')
+  async onDisplayedComponentChanged () {
+    const componentName = newReqModule.displayedComponent
+    if (['SubmissionTabs'].indexOf(componentName) > -1) {
+      await this.startNewNrTimer()
+      this.$set(this, 'displayTimer', true)
```

Comment:
As far as I know, `this.$set` is to add properties to a reactive object.
If I understand correctly, this is not the case here... `displayTimer` is a class property and simply updating it should update the prop in the countdown-timer element.
If I misunderstand, please explain, thanks!

Reply:
this.$set from within a component allows you set data from within a 'method' function of a Vue component. This would be more clear using standard Vue syntax as opposed to the component decorators.
https://stackoverflow.com/questions/51296654/vuejs-set-data-in-methods-function

Reply:
Thanks for the explanation. That SO post is specifically for adding/changing an object's properties, as explained here:
https://www.telerik.com/blogs/so-what-actually-is-vue-set

As for an object watcher, it only fires when the object itself is reassigned. To watch for object property changes you have to use this:
```
@Watch('myObject', { deep: true })
```

Leave it for now but I think it's tech debt related to mixed use of object style vs component style. AFAIK the Entity UIs have not needed to use `this.$set` for any reason.